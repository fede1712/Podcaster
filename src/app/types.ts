export interface Podcast {
  "im:name": { label: string };
  "im:image": image[];
  summary: { label: string };
  "im:price": { label: string; attributes: { amount: string; currency: string } };
  "im:contentType": { attributes: { term: string; label: string } };
  rights: { label: string };
  title: { label: string };
  link: { attributes: { rel: string; type: string; href: string } };
  id: { label: string; attributes: { "im:id": string } };
  "im:artist": { label: string; attributes: { href: string } };
  category: { attributes: { "im:id": string; term: string; scheme: string; label: string } };
  "im:releaseDate": { label: string; attributes: { label: string } };
}

export type image = { label: string; attributes: { height: string } };

export interface Episode {
  country: string;
  artworkUrl160: string;
  episodeFileExtension: string;
  episodeContentType: string;
  artworkUrl600: string;
  collectionViewUrl: string;
  trackTimeMillis: number;
  shortDescription: string;
  closedCaptioning: string;
  collectionId: number;
  collectionName: string;
  genres: genres[];
  episodeGuid: string;
  description: string;
  releaseDate: string;
  previewUrl: string;
  trackId: number;
  trackName: string;
  feedUrl: string;
  artistIds: [];
  episodeUrl: string;
  artworkUrl60: string;
  contentAdvisoryRating: string;
  trackViewUrl: string;
  kind: string;
  wrapperType: string;
  genreIds: string[];
  primaryGenreName: string;
  currency: string;
  trackCount: number;
  trackExplicitness: string;
  artistName: string;
  collectionCensoredName: string;
  trackCensoredName: string;
  artworkUrl30: string;
  artworkUrl100: string;
  collectionPrice: number;
  trackPrice: number;
  collectionHdPrice: number;
  collectionExplicitness: string;
}

export type genres = { name: string; id: string };
