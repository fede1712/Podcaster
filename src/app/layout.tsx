import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "./globals.css";
import Link from "next/link";
import { Header } from "./ui/header/header";

export const metadata: Metadata = {
  title: "Podcaster",
  description: "Itunes top podcasts",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className="m-8">
        <Header />
        {children}
      </body>
    </html>
  );
}
