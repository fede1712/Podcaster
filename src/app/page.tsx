import { getPodcasts } from "./lib/data";
import { PodcastCard } from "./ui/podcastCard/podcastCard";

export default async function Home() {
  const data = await getPodcasts();
  console.log(data.feed.entry[1]);

  return (
    <div className="grid grid-cols-4 gap-14 m-14">
      {data.feed.entry.map((podcast: any) => {
        return <PodcastCard author={podcast["im:artist"].label} title={podcast["im:name"].label} src={podcast["im:image"][2].label} />;
      })}
    </div>
  );
}
