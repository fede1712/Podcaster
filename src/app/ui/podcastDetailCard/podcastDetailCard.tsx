import { Episode } from "@/app/types";
import Image from "next/image";
import Link from "next/link";

export const PodcastDetailCard = ({ data, summary, podcastId }: { data: Episode; summary: string; podcastId: string }) => {
  return (
    <div className="border shadow-xl" data-testid="podcast-detail-card">
      {data && (
        <>
          <div className="flex justify-center border-b-2 p-8">
            <Link href={`/podcast/${podcastId}`}>
              <Image alt={data.collectionName} src={data.artworkUrl600} width={200} height={200} priority={true} />
            </Link>
          </div>
          <section className="p-8 border-b-2">
            <Link href={`/podcast/${podcastId}`}>
              <h3 className="font-bold">{data.collectionName}</h3>
            </Link>
            <p className="italic">by {data.artistName}</p>
          </section>
          <section className="p-8">
            <h3 className="font-bold">Description:</h3>
            <p className="text-sm text-left italic break-words">{summary}</p>
          </section>
        </>
      )}
    </div>
  );
};
