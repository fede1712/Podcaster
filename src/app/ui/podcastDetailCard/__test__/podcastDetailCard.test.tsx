import React from "react";
import { render, screen } from "@testing-library/react";
import { PodcastDetailCard } from "../podcastDetailCard";
import { Episode } from "@/app/types";

describe("Renders PodcastDetailCard correctly", () => {
  test("should render PodcastDetailCard", () => {
    const episode: Episode = {
      trackName: "Test Episode",
      description: "This is a test episode description.",
      episodeUrl: "http://example.com/test-episode.mp3",
      artworkUrl600: "http://example.com/artwork.jpg",
      trackTimeMillis: 123456,
      artistName: "Test artist name",
      collectionName: "Test collection name",
      releaseDate: "Test realease date",
      trackId: 123456,
    };

    render(<PodcastDetailCard data={episode} summary="Podcast summary test" podcastId="1" />);
    const collectionNameElement = screen.queryByText("Test collection name");
    expect(collectionNameElement).not.toBeNull();

    const artistNameElement = screen.queryByText("by Test artist name");
    expect(artistNameElement).not.toBeNull();

    const descriptionElement = screen.queryByText("Description:");
    expect(descriptionElement).not.toBeNull();

    const summaryElement = screen.queryByText("Podcast summary test");
    expect(summaryElement).not.toBeNull();

    const { asFragment } = render(<PodcastDetailCard data={episode} summary="Podcast summary test" podcastId="1" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
