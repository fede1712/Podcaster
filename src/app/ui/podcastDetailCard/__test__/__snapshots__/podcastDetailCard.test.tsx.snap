// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders PodcastDetailCard correctly should render PodcastDetailCard 1`] = `
<DocumentFragment>
  <div
    class="border shadow-xl "
  >
    <div>
      <div
        class="flex justify-center border-b-2 p-8"
      >
        <a
          href="/podcast/1"
        >
          <img
            alt="Test collection name"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="200"
            src="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg&w=640&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </a>
      </div>
      <section
        class="p-8 border-b-2"
      >
        <a
          href="/podcast/1"
        >
          <h3
            class="font-bold"
          >
            Test collection name
          </h3>
        </a>
        <p
          class="italic"
        >
          by Test artist name
        </p>
      </section>
      <section
        class="p-8"
      >
        <h3
          class="font-bold"
        >
          Description:
        </h3>
        <p
          class="text-sm text-left italic break-words"
        >
          Podcast summary test
        </p>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
